crowdsourcing/viewsets/task.py

```diff
@@ -261,3 +284,37 @@ class CurrencyViewSet(viewsets.ModelViewSet):
 
     queryset = Currency.objects.all()
     serializer_class = CurrencySerializer
+
+
+class ExternalSubmit(APIView):
+    def post(self, request, *args, **kwargs):
+        identifier = request.GET.get('daemo_id', False)
```

Comment:
What if they send multiple AJAX requests? Do we append to the data we
record? Or something else?

On Saturday, December 19, 2015, Durim Morina <notifications@github.com>
wrote:

> In crowdsourcing/viewsets/task.py
> <https://github.com/crowdresearch/crowdsource-platform/pull/604#discussion_r48096791>
> :
>
> > @@ -261,3 +284,37 @@ class CurrencyViewSet(viewsets.ModelViewSet):
> >
> >      queryset = Currency.objects.all()
> >      serializer_class = CurrencySerializer
> > +
> > +
> > +class ExternalSubmit(APIView):
> > +    def post(self, request, *args, **kwargs):
> > +        identifier = request.GET.get('daemo_id', False)
>
> if we listen on $location change of the iFrame as we discussed and we
> receive a submission from their page then we can mark it as submitted,
> workflow doesn't change from the regular POST
> workflow is: 1. they submit the form data whichever way they prefer (ajax
> or regular POST) and 2. they either let us know the user is done by
> location change to something like /task-done and we capture that info and
> continue from there
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/crowdresearch/crowdsource-platform/pull/604/files#r48096791>
> .
>


-- 
sent from my NLS